types.go

```diff
@@ -177,21 +176,22 @@ func newString(ctx context, m *Meta, s string) *cfgString {
 	return &cfgString{cfgPrimitive{ctx, m}, s}
 }
 
-func newRef(ctx context, m *Meta, ref *reference) (*cfgDynamic, error) {
+func newRef(ctx context, m *Meta, ref *reference) *cfgDynamic {
 	return newDyn(ctx, m, (*refDynValue)(ref))
 }
 
-func newSplice(ctx context, m *Meta, s varEvaler) (*cfgDynamic, error) {
+func newSplice(ctx context, m *Meta, s varEvaler) *cfgDynamic {
 	return newDyn(ctx, m, spliceDynValue{s})
 }
 
-func newDyn(ctx context, m *Meta, val dynValue) (*cfgDynamic, error) {
-	idElem, err := uuid.NewV4()
-	if err != nil {
-		return nil, err
-	}
-	id := string(atomic.AddInt32(&spliceSeq, 1)) + idElem.String()
-	return &cfgDynamic{cfgPrimitive{ctx, m}, cacheID(id), val}, nil
+func newDyn(ctx context, m *Meta, val dynValue) *cfgDynamic {
+	dyn := new(cfgDynamic)
+	randID := fmt.Sprintf("%p-%d", &dyn, time.Now().Unix())
+	id := string(atomic.AddInt32(&spliceSeq, 1)) + randID
```

Comment:
This will result in an extra allocation. Let's not create a temporary string. Plus this change allocates dyn and the takes the address of the dyn variable, which would give the stack-address, not the heap-address.

How about:

```
seq := atomic.AddInt32(&spliceSeq, 1)
dyn := &cfgDynamic{cfgPrimitive: cfgPrimitive{ctx, m}, dyn: val}
dyn.id := fmt.Sprintf("%8X-%4X-%p", time.Now.Unix(), seq, dyn)
return dyn
```

Note: in go-ucfg prefer initializer syntax over new/make keyword. E.g. `dyn := &cfgDynamic{}` over `new(cfgDynamic)`. Only use make for channels or if size for map/slice/channel is known when initializing a variable.